### PR TITLE
Added support for all platforms in the `oryx run-script` command

### PR DESCRIPTION
**Pull request by dorfire**

### CODE REVIEW

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.BuildScriptGenerator.Exceptions;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    internal class DefaultRunScriptGenerator : IRunScriptGenerator
```
Comment (dorfire) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs]: This should be showing as a rename: https://github.com/microsoft/Oryx/pull/194/commits/f49586d3346b672127cde8fbe39ba349365e61ae

File: src/BuildScriptGenerator/Contracts/IProgrammingPlatform.cs [hunk 2]
```diff
@@ -62,13 +62,6 @@ public interface IProgrammingPlatform
         /// <c>null</c> otherwise.</returns>
         BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorContext scriptGeneratorContext);
 
-        /// <summary>
-        /// Generate a bash script that can run the application.
-        /// </summary>
-        /// <param name="runScriptGeneratorOptions">Options to generate the run script with.</param>
-        /// <returns>The bash run script.</returns>
-        string GenerateBashRunScript(RunScriptGeneratorOptions runScriptGeneratorOptions);
```
Comment (dorfire) [inline src/BuildScriptGenerator/Contracts/IProgrammingPlatform.cs]: @gpcastro

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        private const string TempScriptPath = "/tmp/run.sh";
```
Comment (gpcastro) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs]: Use `System.IO.Path.GetTempPath()` instead

File: src/BuildScriptGenerator/Node/NodePlatform.cs [hunk 4]
```diff
@@ -223,93 +223,6 @@ public void SetVersion(BuildScriptGeneratorContext context, string version)
...
-            _logger.LogInformation("Finalizing entrypoint script using {commandSource}", commandSource);
-            var templateValues = new NodeBashRunScriptProperties
-            {
-                AppDirectory = options.SourceRepo.RootPath,
-                StartupCommand = startupCommand,
-                ToolsVersions = string.IsNullOrWhiteSpace(options.PlatformVersion)
-                ? null : $"node={options.PlatformVersion}",
-                BindPort = options.BindPort
-            };
-            var script = TemplateHelpers.Render(TemplateHelpers.TemplateResource.NodeRunScript, templateValues);
```
Comment (gpcastro) [inline src/BuildScriptGenerator/Node/NodePlatform.cs]: you can delete the resource as well (tpl file) and the entry with its name from the resource file

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        private readonly IEnumerable<IProgrammingPlatform> _programmingPlatforms;
+        private readonly ILogger<DefaultRunScriptGenerator> _logger;
+
+        public DefaultRunScriptGenerator(IEnumerable<IProgrammingPlatform> platforms, ILogger<DefaultRunScriptGenerator> logger)
+        {
+            _programmingPlatforms = platforms;
+            _logger = logger;
+        }
+
+        public string GenerateBashScript(string targetPlatformName, RunScriptGeneratorOptions opts)
```
Comment (gpcastro) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs]: can you please add tests?

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    internal class DefaultRunScriptGenerator : IRunScriptGenerator
+    {
+        private const string TempScriptPath = "/tmp/run.sh";
+
+        private readonly IEnumerable<IProgrammingPlatform> _programmingPlatforms;
+        private readonly ILogger<DefaultRunScriptGenerator> _logger;
+
+        public DefaultRunScriptGenerator(IEnumerable<IProgrammingPlatform> platforms, ILogger<DefaultRunScriptGenerator> logger)
```
Comment (gpcastro) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs]: I'm not seeing changes to the CLI... how this be called?

Comment (gpcastro) [reply src/BuildScriptGenerator/DefaultRunScriptGenerator.cs]: Let's allow the user to specify the output file (like the actual startup gens in the run images) as opposed to just print them in the stdout. That pattern allows us to print logs of what we're doing to the stdout.

Comment (dorfire) [reply src/BuildScriptGenerator/DefaultRunScriptGenerator.cs]: it's already like that - there's a `--output` option

**Pull request merged**